### PR TITLE
Compose secrets

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: images/compose/compose.folders.yml [hunk 1]
```diff
@@ -1,30 +1,46 @@
-version: '3'
+# This file is intended as a starting point. Before using it, make a copy and adjust at least the passwords.
```
Comment (gerrod3) [inline images/compose/compose.folders.yml]: Shouldn't this comment make reference to the new password files? It kind of reads like the passwords should be changed here.

File: images/compose/compose.folders.yml [hunk 2]
```diff
@@ -1,30 +1,46 @@
...
+---
+version: '3.7'
```
Comment (gerrod3) [inline images/compose/compose.folders.yml]: Is there any reason why the version is now 3.7?

File: images/compose/compose.folders.yml [hunk 3]
```diff
@@ -42,16 +61,23 @@ services:
...
     depends_on:
       redis:
+        condition: service_healthy
       postgres:
         condition: service_healthy
     hostname: pulp-api
     user: pulp
     volumes:
       - "./assets/settings.py:/etc/pulp/settings.py:z"
-      - "./assets/certs:/etc/pulp/certs:z"
```
Comment (gerrod3) [inline images/compose/compose.folders.yml:106]: Why are you removing the certs directory? Isn't this crucial for setting up cert auth and the CertGuard plugin?

File: images/compose/compose.folders.yml [hunk 4]
```diff
@@ -76,13 +109,31 @@ services:
...
+      - db_encryption_key
+    networks:
+      - pulp_internal
+    restart: always
+
+networks:
+  pulp_internal:
+
+secrets:
+  app_secret:
```
Comment (gerrod3) [inline images/compose/compose.folders.yml]: Very cool use of compose features!

File: images/compose/compose.yml
Comment (gerrod3) [inline images/compose/compose.yml:1]: What is suppose to be the difference between this file and the compose.folders.yml file?

File: images/compose/compose.folders.yml (hunk 2 above)
Comment (mdellweg) [reply images/compose/compose.folders.yml]: Not part of this pr anymore...
Also not a particular reason. Just from examples.

File: images/compose/compose.folders.yml (hunk 3 above)
Comment (mdellweg) [reply images/compose/compose.folders.yml:106]: Cert auth, I'm not sure, but certguard will have all the certs in the database.

Comment (mdellweg) [reply images/compose/compose.folders.yml:106]: Would they auth and webserver certs not be in /etc/pki ?

Comment (mdellweg) [reply images/compose/compose.folders.yml:106]: Also tls should be terminated by nginx and not pulp-api.

File: images/compose/compose.yml
Comment (mdellweg) [reply images/compose/compose.yml:1]: compose.folders is using directories on the host instead of volumes. That's why the diff should be very small.

**Comment by mdellweg** [comment]
This is not the PR you are looking for, StaleBot!

**Pull request closed without merging**